### ISSUE TITLE
Doc is missing

### ISSUE THREAD

**Issue by FerreolS** [JMMC-OpenDev/oimaging#11]
Should we add button displaying the doc for the different algorithm or just rely on an external website?

**Comment by buthanoid** [comment]
My point of view is : 
- nothing can be more complete and efficient than an external website, so obviously we should not aim at being perfect and always include a link to an external website. 
- a live accurate sign concerning keywords for example, is very efficient, for example take IDE that displays information about parameters, their types, and short description of a method when you hover it with the mouse. It is incomplete but very efficient. Is such thing would be useful for algorithms parameters for example, it would be a good functionality to add. Same would be for displaying a average-sized description of the algorithm. Something between the "memory aid" and a synthetic explanation of the algorithm.
- if we are aiming at very beginners users, this cannot replace a demo with either a video, either a "guided GUI tour" with popups indicating the user the current step and what to do. This is a lot of work.

**Comment by FerreolS** [comment]
After discussion with Antoine, we propose to add an help button near the algorithm menu. This will show a popup with a brief summary of the method, the description of (and only) the keyword accessible through the GUI, and a link to the algorithm webpage

**Comment by bourgesl** [comment]
Great, you could add authors, contact email ? maybe citation / license terms ?

Please create 1 text document per software in the doc/ folder (markdown, plain ASCII or html ?), anyway we'll manage how to parse & display them in the GUI.

**Comment by buthanoid** [comment]
Decision needs to be taken about choosing between the two following alternatives:
1) the help button opens a tab in the user browser, to a web page containing documentation
2) the help button opens a dialog in OImaging, displaying documentation, and a link to a more complete web page

If we choose Option 1), we can directly host the markdown file on github, either in the repository, or in the "wiki" pages. Github will already display "beautifully" the file, with CSS and stuff.

If we choose Option 2), we need to remark than choosing Markdown have some consequences. A markdown parser needs to be added to the project, and it is not as small as someone would expect for a "simple" markup format. Moreover, parsing the same file, at every run of OImaging, is a waste of computation resource. Considering than the documentation will (IMHO) hardly be rewritten more than once a year, parsing the Markdown once, at the time of writing, and storing it in another format (i.e HTML), will permit to avoid the parsing. Also, modifying an HTML file, just for updating an hyperlink for example, is not difficult for the maintainer of the documentation (all the tedious part have been done at the initial parse). 

Also, the Swing simple HTML displayer is a bit outdated ; AMHO it is sufficient for timeless tags like lists and stuff. But Laurent has proposed to use ASCII art which could be perfectly sufficient for a simple documentation dialog.

**Comment by FerreolS** [comment]
I think that opening a tab in the browser is often annoying but I everyone think it is a good solution, I'm fine.
For the solution 2, is it possible to parse the md only during building? I think a short markdown doc is really easy to maintain.  Probably we can also use such a dialog for tutorial.

**Comment by bourgesl** [comment]
I agree with ferreol that markdown files are the reference files maintained by scientists in /doc folder.
We could try converting such files to HTML (v3 for swing html viewer) or plain ASCII manually using a shell script, that can be easily displayed in the OImaging GUI (simple): to be tested.

**Comment by bourgesl** [comment]
I used https://markdowntohtml.com/ to convert mira.md to mira.html (added html/body tags) and it looks good:
![mira html_198](https://user-images.githubusercontent.com/5607073/140045371-4f4c381e-2a5c-47fb-b330-7b82da8c28b0.png)
```
        final String doc = "mira.html";
        final String html = FileUtils.readFile(new File(TEST_FOLDER, doc));

        ResizableTextViewFactory.createHtmlWindow(html, doc, true);
```

**Comment by bourgesl** [comment]
@FerreolS please write the other markdown files in /doc.

**Comment by buthanoid** [comment]
- adding jButton in the GUI, at the right of combobox of softwares
- adding documentation:String field in Service class, initialized to null
- adding getDocumentation() function in Service class, return documentation field, but if null, previously try to read a file named after the service program field (like "mira-ci.html") in the resources folder.
- there is the class fr.jmmc.jmcs.util.ResourceUtils with methods to load resources
- there is the class fr.jmmc.jmcs.gui.component.ResizableTextViewFactory made for displaying a window with html content
- markdown files are to be placed in "src/main/resources/fr/jmmc/oimaging/resource/software-documentation/" folder
- html files are to be placed in the same folder, same name but ".md" extension replaced by ".html"
- when updating a mardown file, the programmer must manually convert with ttps://markdowntohtml.com/ and send a commit with both markdown and html updates

**Comment by bourgesl** [comment]
Documents (md html) are stored in the doc folder:
https://github.com/JMMC-OpenDev/oimaging/tree/master/doc

Maven build script will copy these files at the right place in the jar file.

**Comment by FerreolS** [comment]
> when updating a mardown file, the programmer must manually convert with ttps://markdowntohtml.com/ and send a commit with both markdown and html updates

I strongly disagree with manually converting md to html. There is too much risk of discrepancies between both. It must be done during build. For example using: 
`pandoc --standalone  doc.md`

**Comment by bourgesl** [comment]
As you are responsible for doc, please pusg md / html in the same commit.

**Comment by FerreolS** [comment]
No! It has to be done automatically. We should not have to maintain doc on 2 file-format at the same time. Just adding  a line with pandoc at the build seems manageable.

**Comment by bourgesl** [comment]
Please do.

**Comment by bourgesl** [comment]
See github urls:
https://docs.github.com/en/rest/reference/markdown

**Comment by FerreolS** [comment]
Do you mean that you prefer a github action to automatically translate the md in html? That could be a solution. Please be clearer on what are the issue you see with automatic translation.

**Comment by bourgesl** [comment]
I just wanted to not waste my time on writing shell or build script, but I did:
See /doc

**Comment by FerreolS** [comment]
Super. Thank you Laurent

**Comment by buthanoid** [comment]
https://github.com/JMMC-OpenDev/oimaging/pull/45

Did not included yet your new documentation. I don't have time today anymore.

I used a bash script because it permit to not fail compile phase in case pandoc is not available. I assumed bash should be available more often.

**Comment by buthanoid** [comment]
I added a profile in the POM to prevent activation of the plugin calling parsing bash script.
Because if you are on windows, or if you don't have /bin/bash, the plugin fail to complete, and make the whole compile phase to fail, and I did not found how to prevent this. My best try was then to prevent the activation of the plugin if conditions are not met.

Note that the copying of the HTML files to the resources folder is always working because it relies on a well known maven utility that should be very portable.
